src/nvim/os/job.c

```diff
@@ -117,63 +112,45 @@ Job *job_start(JobOptions opts, int *status)
   job->refcount = 1;
   job->stopped_time = 0;
   job->term_sent = false;
-  job->in = NULL;
-  job->out = NULL;
-  job->err = NULL;
   job->opts = opts;
   job->closed = false;
-
-  process_init(job);
-
-  if (opts.writable) {
-    handle_set_job((uv_handle_t *)job->proc_stdin, job);
-    job->refcount++;
-  }
-
-  if (opts.stdout_cb) {
-    handle_set_job((uv_handle_t *)job->proc_stdout, job);
-    job->refcount++;
-  }
-
-  if (opts.stderr_cb) {
-    handle_set_job((uv_handle_t *)job->proc_stderr, job);
-    job->refcount++;
-  }
+  job->in.closed = true;
+  job->out.closed = true;
+  job->err.closed = true;
 
   // Spawn the job
   if (!process_spawn(job)) {
-    if (opts.writable) {
-      uv_close((uv_handle_t *)job->proc_stdin, close_cb);
+    if (job->opts.writable) {
+      uv_close((uv_handle_t *)job->proc_stdin, NULL);
     }
-    if (opts.stdout_cb) {
-      uv_close((uv_handle_t *)job->proc_stdout, close_cb);
+    if (job->opts.stdout_cb) {
+      uv_close((uv_handle_t *)job->proc_stdout, NULL);
     }
-    if (opts.stderr_cb) {
-      uv_close((uv_handle_t *)job->proc_stderr, close_cb);
+    if (job->opts.stderr_cb) {
+      uv_close((uv_handle_t *)job->proc_stderr, NULL);
     }
     process_close(job);
     event_poll(0);
```

Comment:
This should no longer be a problem, the process memory is now managed by the caller